src/test/java/io/jenkins/pluginhealth/scoring/probes/LastCommitDateProbeTest.java

```diff
@@ -0,0 +1,61 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.time.ZonedDateTime;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.model.ResultStatus;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class LastCommitDateProbeTest {
+    @Test
+    public void shouldKeepScmAsKey() {
+        assertThat(new LastCommitDateProbe().key()).isEqualTo("last-commit-date");
+    }
+
+    @Test
+    public void shouldNotRequireRelease() {
+        assertThat(new LastCommitDateProbe().requiresRelease()).isFalse();
+    }
+
+    @Test
+    public void shouldReturnSuccessStatusOnValidSCM() {
+        final Plugin plugin = new Plugin("parameterized-trigger", "https://github.com/jenkinsci/parameterized-trigger-plugin.git", ZonedDateTime.now())
+            .addDetails(ProbeResult.success("scm", "The plugin SCM link is valid"));
```

Comment:
Does this means we are cloning an external repository? What happens if the repository change? If, during the tests execution, there is no connection to internet? Or that GitHub is down?

Reply:
These points helped me understand why we even bother mocking the classes, to remove as many dependencies as possible, and just test the given class in question.